addon/src/services/page-title.ts

```diff
@@ -3,29 +3,60 @@ import { scheduleOnce } from '@ember/runloop';
 import Service, { inject as service } from '@ember/service';
 import { isEmpty } from '@ember/utils';
 import { assert } from '@ember/debug';
+import RouterService from '@ember/routing/router-service';
+import SimpleDomDocument from '@simple-dom/document';
 
-let isFastBoot = typeof FastBoot !== 'undefined';
+const isFastBoot = typeof FastBoot !== 'undefined';
 
 const RouterEvent = {
   ROUTE_DID_CHANGE: 'routeDidChange',
-};
+} as const;
+const configKeys: (keyof PageTitleConfig)[] = [
+  'separator',
+  'prepend',
+  'replace',
+];
+
+export interface PageTitleToken {
+  id: string;
+  title?: string;
+  separator?: string;
+  prepend?: boolean;
+  replace?: boolean;
+  front?: unknown;
+  previous?: PageTitleToken | null;
+  next?: PageTitleToken | null;
+}
+
+interface PageTitleConfig {
+  // The default separator to use between tokens.
+  separator?: string;
+
+  // The default prepend value to use.
+  prepend?: boolean;
+
+  // The default replace value to use.
+  replace?: boolean | null;
+}
```

Comment:
You should explicitly relate these, given it's not safe to let them get out of sync!

```ts
interface PageTitleToken extends PageTitleConfig {
  // ...
}
```

You might also consider extracting and defining the helper options in terms of a dedicated interface, rather than using `Pick`. And if you do that, you should *definitely* use doc-style comments instead of `//`-style comments, so that they get provided to end users of Glint etc.!

```ts
interface PageTitleEnvConfig {
   /** The default separator to use between tokens. */
   separator?: string;

   /** The default prepend value to use. */
   prepend?: boolean;

   /** The default replace value to use. */
   replace?: boolean | null;
}

/** Options users can invoke the helper/function with */
export interface PageTitleRuntimeOptions extends {
  /** If the token should always be in the beginning of the resulting title. */
  front?: boolean;
}

interface PageTitleToken extends PageTitleRuntimeOptions {
  id: string;
  previous?: PageTitleToken | null;
  next?: PageTitleToken | null;
}
```

Reply:
Is `PageTitleRuntimeOptions` the type used by the helper? I'm assuming it should have more options?